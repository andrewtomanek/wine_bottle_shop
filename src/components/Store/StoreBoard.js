import React from "react";
import classes from "./StoreBoard.module.css";
import StoreCard from "./StoreCard";

const StoreBoard = (props) => {
  console.log(props.inventory);
  console.log(props.listNames);
  let transformedItems = Object.keys(props.inventory)
    .map((igKey) => {
      return [...Array(props.inventory[igKey])].map((_, i) => {
        return (
          <StoreCard
            key={igKey + i}
            itemType={igKey}
            itemName={props.listNames[igKey]}
          />
        );
      });
    })
    .reduce((arr, el) => {
      return arr.concat(el);
    }, []);

  console.log(transformedItems);

  if (transformedItems.length === 0) {
    transformedItems = <p className={classes.Empty}>Žádné položky</p>;
  }
  return <div className={classes.Store}>{transformedItems}</div>;
};

export default StoreBoard;
