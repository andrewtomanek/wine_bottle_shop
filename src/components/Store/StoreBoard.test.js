import React from "react";
import { shallow } from "enzyme";
import StoreBoard from "./StoreBoard";

const props = {
  inventory: {
    shopItem1: 1,
    shopItem2: 2,
    shopItem3: 3,
    shopItem4: 4,
    shopItem5: 5,
    shopItem6: 6,
    shopItem7: 7,
    shopItem8: 8,
  },
  listNames: {
    shopItem1: "Solaris",
    shopItem2: "Dornfelder",
    shopItem3: "Agni",
    shopItem4: "Muscaris",
    shopItem5: "Sauvignon",
    shopItem6: "Tramín",
    shopItem7: "Hibernal",
    shopItem8: "Pálava",
  },
};

const emptyProps = { inventory: {} };

describe("Shallow Render StoreBoard component", () => {
  it("renders empty cart", () => {
    const wrapper = shallow(<StoreBoard {...emptyProps} />);
    expect(wrapper.find(".Store").text()).toContain("Žádné položky");
  });

  it("renders shop items", () => {
    const wrapper = shallow(<StoreBoard {...props} />);
    expect(wrapper.find(".Store").children()).toHaveLength(36);
  });

  it("Match Snapshot of StoreBoard component", () => {
    const wrapper = shallow(<StoreBoard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
