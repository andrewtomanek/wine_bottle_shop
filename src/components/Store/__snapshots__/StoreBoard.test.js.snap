// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Render StoreBoard component Match Snapshot of StoreBoard component 1`] = `
<div
  className="Store"
>
  <StoreCard
    itemName="Solaris"
    itemType="shopItem1"
    key="shopItem10"
  />
  <StoreCard
    itemName="Dornfelder"
    itemType="shopItem2"
    key="shopItem20"
  />
  <StoreCard
    itemName="Dornfelder"
    itemType="shopItem2"
    key="shopItem21"
  />
  <StoreCard
    itemName="Agni"
    itemType="shopItem3"
    key="shopItem30"
  />
  <StoreCard
    itemName="Agni"
    itemType="shopItem3"
    key="shopItem31"
  />
  <StoreCard
    itemName="Agni"
    itemType="shopItem3"
    key="shopItem32"
  />
  <StoreCard
    itemName="Muscaris"
    itemType="shopItem4"
    key="shopItem40"
  />
  <StoreCard
    itemName="Muscaris"
    itemType="shopItem4"
    key="shopItem41"
  />
  <StoreCard
    itemName="Muscaris"
    itemType="shopItem4"
    key="shopItem42"
  />
  <StoreCard
    itemName="Muscaris"
    itemType="shopItem4"
    key="shopItem43"
  />
  <StoreCard
    itemName="Sauvignon"
    itemType="shopItem5"
    key="shopItem50"
  />
  <StoreCard
    itemName="Sauvignon"
    itemType="shopItem5"
    key="shopItem51"
  />
  <StoreCard
    itemName="Sauvignon"
    itemType="shopItem5"
    key="shopItem52"
  />
  <StoreCard
    itemName="Sauvignon"
    itemType="shopItem5"
    key="shopItem53"
  />
  <StoreCard
    itemName="Sauvignon"
    itemType="shopItem5"
    key="shopItem54"
  />
  <StoreCard
    itemName="Tramín"
    itemType="shopItem6"
    key="shopItem60"
  />
  <StoreCard
    itemName="Tramín"
    itemType="shopItem6"
    key="shopItem61"
  />
  <StoreCard
    itemName="Tramín"
    itemType="shopItem6"
    key="shopItem62"
  />
  <StoreCard
    itemName="Tramín"
    itemType="shopItem6"
    key="shopItem63"
  />
  <StoreCard
    itemName="Tramín"
    itemType="shopItem6"
    key="shopItem64"
  />
  <StoreCard
    itemName="Tramín"
    itemType="shopItem6"
    key="shopItem65"
  />
  <StoreCard
    itemName="Hibernal"
    itemType="shopItem7"
    key="shopItem70"
  />
  <StoreCard
    itemName="Hibernal"
    itemType="shopItem7"
    key="shopItem71"
  />
  <StoreCard
    itemName="Hibernal"
    itemType="shopItem7"
    key="shopItem72"
  />
  <StoreCard
    itemName="Hibernal"
    itemType="shopItem7"
    key="shopItem73"
  />
  <StoreCard
    itemName="Hibernal"
    itemType="shopItem7"
    key="shopItem74"
  />
  <StoreCard
    itemName="Hibernal"
    itemType="shopItem7"
    key="shopItem75"
  />
  <StoreCard
    itemName="Hibernal"
    itemType="shopItem7"
    key="shopItem76"
  />
  <StoreCard
    itemName="Pálava"
    itemType="shopItem8"
    key="shopItem80"
  />
  <StoreCard
    itemName="Pálava"
    itemType="shopItem8"
    key="shopItem81"
  />
  <StoreCard
    itemName="Pálava"
    itemType="shopItem8"
    key="shopItem82"
  />
  <StoreCard
    itemName="Pálava"
    itemType="shopItem8"
    key="shopItem83"
  />
  <StoreCard
    itemName="Pálava"
    itemType="shopItem8"
    key="shopItem84"
  />
  <StoreCard
    itemName="Pálava"
    itemType="shopItem8"
    key="shopItem85"
  />
  <StoreCard
    itemName="Pálava"
    itemType="shopItem8"
    key="shopItem86"
  />
  <StoreCard
    itemName="Pálava"
    itemType="shopItem8"
    key="shopItem87"
  />
</div>
`;
